internal/providers/pluginfw/resources/catalog/resource_function.go

```diff
@@ -0,0 +1,226 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/retries"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	pluginfwcommon "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/common"
+	pluginfwcontext "github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/context"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/converters"
+	"github.com/databricks/terraform-provider-databricks/internal/providers/pluginfw/tfschema"
+	"github.com/databricks/terraform-provider-databricks/internal/service/catalog_tf"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+)
+
+const resourceName = "function"
+
+var _ resource.ResourceWithConfigure = &FunctionResource{}
+
+func ResourceFunction() resource.Resource {
+	return &FunctionResource{}
+}
+
+func waitForFunction(ctx context.Context, w *databricks.WorkspaceClient, funcInfo *catalog.FunctionInfo) diag.Diagnostics {
+	const timeout = 5 * time.Minute
+
+	result, err := retries.Poll[catalog.FunctionInfo](ctx, timeout, func() (*catalog.FunctionInfo, *retries.Err) {
+		attempt, err := w.Functions.GetByName(ctx, funcInfo.FullName)
+		if err != nil {
+			if apierr.IsMissing(err) {
+				return nil, retries.Continue(fmt.Errorf("function %s is not yet available", funcInfo.FullName))
+			}
+			return nil, retries.Halt(fmt.Errorf("failed to get function: %s", err))
+		}
+		return attempt, nil
+	})
+
+	if err != nil {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("failed to create function", err.Error())}
+	}
+
+	*funcInfo = *result
+	return nil
+}
```

Comment:
Do we really need it for functions? I thought that function creation should be fast.

Reply:
I'll bring this up in the office hours.

Reply:
we need waiters for some resources, like cluster, online table, and Vector Search, because they are creating some resources under the hood. But functions are just definitions in UC, so the creation happens immediately and we don't need to wait